### PR TITLE
Package bls12-381.0.3.15

### PR DESCRIPTION
**Pull request by dannywillems**
### `bls12-381.0.3.15`
OCaml binding for bls12-381 from librustzcash

---
* Homepage: https://gitlab.com/dannywillems/ocaml-bls12-381
* Source repo: git+https://gitlab.com/dannywillems/ocaml-bls12-381.git
* Bug tracker: https://gitlab.com/dannywillems/ocaml-bls12-381/issues

---
:camel: Pull-request generated by opam-publish v2.0.2

### CODE REVIEW

File: packages/bls12-381/bls12-381.0.3.15/opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
+opam-version: "2.0"
+synopsis: "OCaml binding for bls12-381 from librustzcash"
+maintainer: "Danny Willems <be.danny.willems@gmail.com>"
+authors: "Danny Willems <be.danny.willems@gmail.com>"
+license: "MIT"
+homepage: "https://gitlab.com/dannywillems/ocaml-bls12-381"
+bug-reports: "https://gitlab.com/dannywillems/ocaml-bls12-381/issues"
+depends: [
+  "conf-rust" {build}
+  "ocaml" {>= "4.07"}
```
Comment (mseri) [inline packages/bls12-381/bls12-381.0.3.15/opam:10]:
```suggestion
  "ocaml" {>= "4.07" | (>= "4.08" & with-test)}
```

Comment (mseri) [reply packages/bls12-381/bls12-381.0.3.15/opam:10]: Tests are using the Option module which was introduced in 4.08

Comment (kit-ty-kate) [reply packages/bls12-381/bls12-381.0.3.15/opam:10]:
```suggestion
  "ocaml" {>= "4.07"}
  "ocaml" {with-test & >= "4.08"}
```
The version using the disjunction will not work in this case as:
* `with-test = true & ocaml = 4.07 ---> {>= "4.07" | (>= "4.08" & true)} ---> {true | false} ---> true`

File: packages/bls12-381/bls12-381.0.3.15/opam [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "dune" {>= "1.7"}
```
Comment (kit-ty-kate) [inline packages/bls12-381/bls12-381.0.3.15/opam:11]: Your `dune-project` file states that this package requires at least dune 2.7, not 1.7

File: packages/bls12-381/bls12-381.0.3.15/opam [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "dune-configurator" {build}
+  "ff" {>= "0.4.0" & < "0.5.0"}
+  "zarith" {>= "1.10" & < "2.0"}
+  "ctypes" {= "0.17.1"}
+  "ctypes-foreign" {= "0.4.0"}
+  "tezos-rust-libs" {= "1.0"}
```
Comment (kit-ty-kate) [inline packages/bls12-381/bls12-381.0.3.15/opam:17]: is there any chance of not using equality constraints here? This is very problematic in opam-repository as it forbids any patches to be used. Could this instead be something like
```
ctypes >= 0.17.1 & < 0.18.0
ctypes-foreign (this one is a meta packages so no need for a constraint)
tezos-rust-libs >= 1.0 & < 1.1 or something
```

File: packages/bls12-381/bls12-381.0.3.15/opam (hunk 1 above)
Comment (mseri) [reply packages/bls12-381/bls12-381.0.3.15/opam:10]: Indeed, well spotted!

Comment (dannywillems) [reply packages/bls12-381/bls12-381.0.3.15/opam:10]: I will restrict to >= 4.08, it is simpler and cleaner.

File: packages/bls12-381/bls12-381.0.3.15/opam (hunk 3 above)
Comment (dannywillems) [reply packages/bls12-381/bls12-381.0.3.15/opam:17]: I relaxed the constraint on `ctypes` as suggested and remove the version constraint on `ctypes-foreign`.
For `tezos-rust-libs`, I prefer to be strict for the moment. I don't know yet the which versioning system that will be used.

File: packages/bls12-381/bls12-381.0.3.15/opam (hunk 2 above)
Comment (dannywillems) [reply packages/bls12-381/bls12-381.0.3.15/opam:11]: The dune-project states it requires 2.0 (this release comes from the branch `match-tezos`, built over `0.3.14` as the master branch contains already some changes not integrated yet in `tezos-rust-libs`). Changing.
This development process will change in the next weeks when `tezos-rust-libs` will integrate the changes required for `bls12-381.0.4.0`.

**Comment by dannywillems** [comment]
Closing in favor of #18073

**Pull request closed without merging**